core/client/src/client.rs

```diff
@@ -260,6 +253,7 @@ impl<H> PrePostHeader<H> {
 pub fn new_in_mem<E, Block, S, RA>(
 	executor: E,
 	genesis_storage: S,
+	keystore: Option<primitives::traits::KeyStorePtr>,
```

Comment:
is there a strong case for putting this in the client? 
a god-object client gave us a lot of trouble in parity-ethereum; we have an opportunity here to avoid making the same mistake.

Reply:
for instance, why can the keystore not be encapsulated as part of the `E` parameter?

Reply:
It is not part of the client.

Reply:
yes it is! it goes into the `LocalCallExecutor` which is part of the client.